src/matcher/util.ts

```diff
@@ -0,0 +1,37 @@
+import { MatcherElement } from "./types";
```

Comment:
what's different between functions in `matcher/utils` and  `utils/fn` ? 

can we put them together? and we need UT for these functions.

Reply:
I'd prefer to put matcher only utils and type in to matcher folder, and observer only to observer folder.
And leave common usage  in to the global utils and type folder.
I think It may help to compile and treeshake.